zyte_parsers/aggregate_rating.py

```diff
@@ -0,0 +1,112 @@
+import re
+from math import isnan
+from typing import Any, List, Optional
+
+import attr
+from lxml.html import HtmlElement
+
+from .api import SelectorOrElement, input_to_element
+from .utils import extract_text
+
+
+@attr.s(frozen=True, auto_attribs=True)
+class AggregateRating:
+    bestRating: Optional[float] = None
+    ratingValue: Optional[float] = None
+
+
+POSSIBLE_BEST_RATINGS = {4.0, 5.0, 6.0, 10.0, 20.0, 100.0}
+
+
+def extract_rating(node: SelectorOrElement) -> AggregateRating:
+    """Extract rating data from a node.
+
+    :param node: Node that includes the rating data.
+    :return: AggregateRating item.
+    """
+    node = input_to_element(node)
+    node_text = extract_text(node)
+    rating_value = None
+    best_rating = None
+    if node_text is None:
+        return AggregateRating(ratingValue=rating_value, bestRating=best_rating)
+    node_nums = _get_rating_numbers(node_text)
+    if len(node_nums) == 2:
+        rating_value = node_nums[0]
+        best_rating = node_nums[1]
```

Comment:
Do we validate it in other code paths? Do you mean `_check_best_rating()`?

Reply:
Yes, _check_best_rating, which is called in the other code path.

Reply:
By the way, I think we shouldn't make any changes to the logic without checking it on the datasets (i.e. without json tests).

Reply:
I wrote a test in zyte-common-items that parses "3.8 (7 reviews)" and it interpreted 7 as bestRating, via this code path. So it would be indeed better to do some validation here.